### PR TITLE
feat: [PL-26264]: get repo url API git sdk integration

### PR DESCRIPTION
**Pull request by mohitg0795**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/34991)

### CODE REVIEW

**Review by namanvermah — APPROVED** [review]
:lgtm:

> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34991#-:-N6Bm_7cEpkSWnU270Lo:bnfp4nl)* status: 4 file(s) to review. 1 of 2 approvals obtained.

File: 900-git-sync-sdk/src/main/java/io/harness/gitsync/scm/SCMGitSyncHelper.java [hunk 1]
```diff
@@ -231,6 +234,28 @@ public ScmCreatePRResponse createPullRequest(Scope scope, String repoName, Strin
     return ScmCreatePRResponse.builder().prNumber(createPRResponse.getPrNumber()).build();
   }
 
+  public ScmGetRepoUrlResponse getRepoUrl(
+      Scope scope, String repoName, String connectorRef, Map<String, String> contextMap) {
```
Comment (bhavya181) [inline 900-git-sync-sdk/src/main/java/io/harness/gitsync/scm/SCMGitSyncHelper.java:238]: Please call method to set log here

**Review by bhavya181 — CHANGES_REQUESTED** [review]

**Comment by mohitg0795** [comment]
*[`900-git-sync-sdk/src/main/java/io/harness/gitsync/scm/SCMGitSyncHelper.java` line 238 at r1](https://reviewable.io/reviews/harness/harness-core/34991#-N69kzwZA9lZ_COIfjX6-r1-238:-N6ZdtJS-n64SVWyyhzI:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/a544d08d4ee744e88c0f5a1ebfad4fbd51abca9b/900-git-sync-sdk/src/main/java/io/harness/gitsync/scm/SCMGitSyncHelper.java#L238)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

Please call method to set log here
</blockquote></details>

done

**Comment by mohitg0795** [comment]
trigger messagecheck

**Review by bhavya181 — APPROVED** [review]
> Reviewed 2 of 4 files at r1, 1 of 2 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34991)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Review by kapil-harness — APPROVED** [review]
> Reviewed 2 of 4 files at r1, 1 of 2 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34991)* status: :shipit: complete! ✓

**Pull request merged**